src/RestServer/Controllers/v1/WalletController.cs

```diff
@@ -0,0 +1,676 @@
+// Copyright (C) 2015-2023 The Neo Project.
+//
+// The Neo.Plugins.RestServer is free software distributed under the MIT software license,
+// see the accompanying file LICENSE in the main directory of the
+// project or http://www.opensource.org/licenses/mit-license.php
+// for more details.
+//
+// Redistribution and use in source and binary forms with or without
+// modifications are permitted.
+
+using Akka.Actor;
+using Microsoft.AspNetCore.Cors;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Neo.Network.P2P.Payloads;
+using Neo.Plugins.RestServer.Exceptions;
+using Neo.Plugins.RestServer.Helpers;
+using Neo.Plugins.RestServer.Models.Error;
+using Neo.Plugins.RestServer.Models.Wallet;
+using Neo.SmartContract;
+using Neo.SmartContract.Native;
+using Neo.Wallets;
+using Neo.Wallets.NEP6;
+using System.Net.Mime;
+using System.Numerics;
+using static Neo.SmartContract.Helper;
+
+namespace Neo.Plugins.RestServer.Controllers.v1
+{
+    /// <summary>
+    /// Wallet API
+    /// </summary>
+    [Route("/api/v{version:apiVersion}/wallet")]
+    [DisableCors]
+    [Produces(MediaTypeNames.Application.Json)]
+    [Consumes(MediaTypeNames.Application.Json)]
+    [ProducesResponseType(StatusCodes.Status400BadRequest, Type = typeof(ErrorModel))]
+    [ApiVersion("1.0")]
+    [ApiController]
+    public class WalletController : ControllerBase
+    {
+        internal static WalletSessionManager WalletSessions { get; } = new();
+
+        private readonly NeoSystem _neosystem;
+
+        /// <summary>
+        /// CTOR
+        /// </summary>
+        /// <exception cref="NodeNetworkException">Node network doesn't match plugins network.</exception>
+        public WalletController()
+        {
+            _neosystem = RestServerPlugin.NeoSystem ?? throw new NodeNetworkException();
+        }
+
+        /// <summary>
+        /// Opens a wallet.
+        /// </summary>
+        /// <param name="model"></param>
+        /// <returns>A newly created wallet session object.</returns>
+        /// <response code="200">Returns newly create wallet session object.</response>
+        /// <response code="400">An error occurred. See Response for details.</response>
+        [HttpPost("open", Name = "WalletOpen")]
+        [ProducesResponseType(StatusCodes.Status200OK, Type = typeof(WalletSessionModel))]
+        public IActionResult WalletOpen(
+            [FromBody]
+            WalletOpenModel model)
+        {
+            if (AppContext.BaseDirectory.StartsWith(new FileInfo(model.Path).DirectoryName, StringComparison.InvariantCultureIgnoreCase) == false)
```

Comment:
`c:\myvalidPath\myValidPath2\..\..\` it will work

Reply:
Good catch. I'll take your recommendation on how to fix. Also if you want you can update this `method`.